producer-kafka/README.adoc

```diff
@@ -0,0 +1,407 @@
+Once the record is created, it is sent to the Apache Kafka topic.
+
+4. `close()` - it flushes all pending records from the underlying Kafka producer and closes it.
+
+image::documentation/Kafka_CDC_architecture.png[Kafka CDC architecture]
+
+Let's recap the basics of architecture. The CDC connector is working within a Stargate node. The stargate node sends the TableMetadata when the schema is changed, OR the Stargate node is restarted.
+It constructs the schema and registers it in the schema registry. When the new CDC event occurs, the Stargate is propagating
+this change to the CDC extension via the send() method. The send() retrieves the schema form schema registry (or a local cache), constructs the Avro record, and sends it to Kafka.
+
+== Configuration
+
+The `init()` method takes the `Map<String, Object>` as an argument. This map must contain the required settings for running the Kafka CDC connector.
+There are multiple required settings:
+
+=== Prefix Name
+
+`cdc.topic.prefix-name` - it should contain a `String` value. It is used to construct the topic name to which the CDC modification for a given TableMetadata is sent.
+Value from this setting is used as a prefix. For example, if the prefix is 'p', keyspace 'ks' and table 't', the final topic name will be: 'p.ks.t'
+The send() will publish a CDC event to a Kafka topic constructed  using this strategy
+
+=== Kafka Producer Settings
+All Kafka Producer settings must be prefixed with `cdc.kafka.producer`.
+The Kafka producer has an extensive list of producer settings. For a full list, please visit the link:https://kafka.apache.org/documentation/#producerconfigs[Kafka producer configs].
+All settings with the `cdc.kafka.producer` passed to this extension will be passed to the Kafka producer.
+So for example, the `cdc.kafka.producer.max.block.ms` will be trimmed to `max.block.ms` and passed to the producer.
+There are two required settings that you need to pass to the CDC connector.
+
+- `cdc.kafka.producer.bootstrap.servers` - please note that you need to prefix it with CDC setting prefix. It contains a list of Kafka servers.
+- `cdc.kafka.producer.schema.registry.url` - it contains one URL or multiple urls to a schema registry that will be used to schema management.
+The `schema-registry` should work in the `full_transitive` compatibility mode to allow the backward and forward compatible evolution of schema.
+To see how to achieve it, please visit the link:https://docs.confluent.io/current/schema-registry/avro.html#compatibility-types[Schema Registry Guide].
+- Two Kafka producer settings cannot be set by the user: `key.serializer` and `value.serializer` - must be set to `KafkaAvroSerializer.class`
+because the CDC producer uses Avro format for publishing events to a Kafka. If the user passes those settings, they will be ignored.
```

Comment:
```suggestion
- There are two Kafka producer settings which cannot be set by the user: `key.serializer` and `value.serializer` - both must be set to `KafkaAvroSerializer.class`
because the CDC producer uses Avro format for publishing events to Kafka. Any value provided for these two settings will be ignored.
```